sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs

```diff
@@ -3871,6 +3879,45 @@ private void SetNameFieldsIfNull()
         #endregion UploadRangeFromUrl
 
         #region Upload
+        /// <summary>
+        /// The <see cref="Upload(Stream, IProgress{long}, ShareFileRequestConditions, CancellationToken)"/>
+        /// operation writes <paramref name="content"/> to a file.
+        ///
+        /// For more information, see
+        /// <see href="https://docs.microsoft.com/rest/api/storageservices/put-range">
+        /// Put Range</see>.
+        /// </summary>
+        /// <param name="content">
+        /// A <see cref="Stream"/> containing the content of the file to upload.
+        /// </param>
+        /// <param name="options">
+        /// Optional parameters.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate notifications
+        /// that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{StorageFileUploadInfo}"/> describing the
+        /// state of the file.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual Response<ShareFileUploadInfo> Upload(
+            Stream content,
+            ShareFileUploadOptions options,
+            CancellationToken cancellationToken = default) =>
+            UploadInternal(
+                content,
+                options?.ProgressHandler,
+                options?.Conditions,
+                options?.TransferOptions.MaximumTransferSize <= int.MaxValue ? (int)options?.TransferOptions.MaximumTransferSize : default,
```

Comment:
I'd rather switch this implementation to `Uploader` altogether. Imo doesn't make sense to let bunch of knobs in that take no effect. Otherwise I'd accept single MaximumTransferSize instead of StorageTransferOptions.

Btw. if this implementation can't take `long` (now or after switching to uploader) then I'd rather validate and throw instead of falling back to some value.

Reply:
Will do.